### DISCUSSION TITLE
systemd-coredump fail: "Failed to connect to coredump service: No such file or directory"

### DISCUSSION THREAD

**Discussion by Marqin** [microsoft/WSL#10364] [Q&A]
Hey, I have updated my config to use systemd (updated WSL, using ubuntu 20.04 image etc etc), and when I try to get coredumps from crashes I'm getting no coredumps in coredumpctl.

I inspected the journal for the service and it's full of entries like this:

```
Failed to connect to coredump service: No such file or directory
```

P.S. im not interested in setting coredump pattern to /tmp/core, I want systemd-coredump to work

**Comment by dawesc** [comment]
I finally found it:
[](https://www.stigviewer.com/stig/red_hat_enterprise_linux_8/2020-11-25/finding/V-230312)

Somebody has disabled the wretched part of systemd that actually runs this; so you need to (on my RHEL8 system) run
```
sudo systemctl unmask systemd-coredump.socket
sudo systemctl restart systemd-coredump.socket
```

**Reply by evening-hs** [reply]
I have the same issue, however when I try to run those commands I get 

```
Failed to get properties: Transport endpoint is not connected
```

**Comment by zcobol** [comment]
By default `systemd-coredump.socket` is in `static` state, meaning it is wanted by another service. It's not recommended to change it's state. On boot it will open the `/run/systemd/coredump` socket that will be used by `/usr/lib/systemd/systemd-coredump`. Run `sysctl kernel.core_pattern` for details.
